pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -2594,7 +2596,22 @@ func parseEnvVariablesFromCloudInit(envPairs []string) (map[string]string, error
 			errStr := fmt.Sprintf("Variable \"%s\" not defined properly\nKey value pair should be delimited by \"=\"", pair[0])
 			return nil, errors.New(errStr)
 		}
-		envList[pair[0]] = pair[1]
+
```

Comment:
IMHO it would be good to explain that you want to fail parsing (while go-envparse is trying to correct), because otherwise you could just do:
```
return envp.Parse(bytes.NewReader([]byte(strings.Join(envPairs, "\n")))
```